Controllers/order.controller.js

```diff
@@ -14,26 +14,60 @@ import razorpay from "../config/razorpay.config.js";
  *********************************************************/
 
 export const generateRazorpayOrderId = asyncHandler( async (req, res)=>{
-    //get product and coupon from frontend
+    // Get Product and Coupon From Frontend
 
-    //verfiy product price from backend
-    // make DB query to get all products and info
+    const { products : Products, code : couponCode} = req.body;
 
-    let totalAmount;
-    //total amount and final amount
-    // coupon check - DB
-    // disount
-    // finalAmount = totalAmount - discount
+    if(!Products){
+        throw new CustomError("No Product was Found",404);
+    };
+
+    
+    const coupon = await Coupon.find({code : { $eq : couponCode } });
+    
+    // Coupon Check DB
+    if(!coupon || !coupon.active){
```

Comment:
what if i don't want to pass an coupon

Reply:
Hi thank u for the suggestion i take i genuinely think u are absolutely right i will work on it and update the changes.

Reply:
> what if i don't want to pass an coupon

hii Code has been Updated.